CommonClient.py

```diff
@@ -72,9 +72,16 @@ def _cmd_disconnect(self) -> bool:
 
     def _cmd_received(self) -> bool:
         """List all received items"""
-        self.output(f'{len(self.ctx.items_received)} received items:')
+        item: NetworkItem
+        self.output(f'{len(self.ctx.items_received)} received items, sorted by time:')
         for index, item in enumerate(self.ctx.items_received, 1):
-            self.output(f"{self.ctx.item_names[item.item]} from {self.ctx.player_names[item.player]}")
+            parts = []
+            add_json_item(parts, item.item, self.ctx.slot, item.flags)
+            add_json_text(parts, " from ")
+            add_json_location(parts, item.location, item.player)
+            add_json_text(parts, " by ")
+            add_json_text(parts, str(item.player), type=JSONTypes.player_id)
```

Comment:
> why cast item.player to a string? it seems to be an int ...
You mean here? This is the `text` parameter, it immediately gets stringified one level down within `add_json_text()`:

```python
def add_json_text(parts: list, text: typing.Any, **kwargs) -> None:
    parts.append({"text": str(text), **kwargs})
```

I don't see an issue with this. Maybe some explicit typing on the `type` parameter and some typing `@overload` magic can make the type-checking better, (ie associating certain types for `text` with certain values of `type`) but I'd consider that out of scope for this PR.

Reply:
I agree, the `str()` should be removed. This now runs `str(str())` and also it's confusing why the player number would be passed in as str if the function takes Any. That being said, I feel like `add_json_player` would be an even better solution because it abstracts away that `text` is used for player, and on the other end we already have an explicit handler for player.